### PR TITLE
Do validation during client construction

### PR DESCRIPTION
**Pull request by blobor**
With this PR we will switch to validation of jwt payload on client creation. Although validation will be done with `joi` library.

### CODE REVIEW

File: index.test.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
+const R = require('ramda')
 const test = require('ava')
 const jsonwebtoken = require('jsonwebtoken')
 const makeClient = require('.')
 
 const payload = {
-  id: 123,
+  id: 'a3087d3e-8a4c-44f2-a279-d3097c5bdd49',
```
Comment (blobor) [inline index.test.js]: In our system services uses `UUID` as user identifier.

**Comment by blobor** [comment]
> ko-search-react is / will use this library, and I don't want to have to do conditional require/imports for joi since the browser client will never need that dependency. I also don't want to make our dependency bundle larger than it already is.

👍

**Comment by blobor** [comment]
@neezer perfectly valid comment, thanks for clarification!

File: index.js [hunk 2]
```diff
@@ -1,47 +1,37 @@
 const jsonwebtoken = require('jsonwebtoken')
 
+const assertShape = payload => {
+  ;['id', 'ort', 'tenantId', 'tenants'].forEach(fieldName => {
```
Comment (neezer) [inline index.js]: What's this leading `;`?

Comment (blobor) [reply index.js]: Whoops

Comment (blobor) [reply index.js]: ok, got it. It's because https://github.com/standard/standard/blob/master/docs/RULES-en.md#semicolons

**Pull request merged**